backend/clickhouse/cursor.go

```diff
@@ -0,0 +1,55 @@
+package clickhouse
+
+import (
+	"encoding/base64"
+
+	"fmt"
+	"strings"
+	"time"
+
+	e "github.com/pkg/errors"
+
+	modelInputs "github.com/highlight-run/highlight/backend/private-graph/graph/model"
+)
+
+func getLogsPayload(logs []*modelInputs.LogEdge, limit uint64) *modelInputs.LogsPayload {
```

Comment:
would it be possible to make this bi-directional? we might have some usecases for infinite scrolling in the other direction as well

Reply:
Yes. I think we should definitely support backwards scrolling and I see no reason the logic can't support it (just flip the operators). 
Let's tackle that when we do #4396